js/base/base.js

```diff
@@ -0,0 +1,64 @@
+function Base(element, nameOfElement) {
+    this.basis = element;
+    this.nameOfElement = nameOfElement;
+
+    const closeBtn = createCloseBtnHTML();
+    this.basis.appendChild(closeBtn);
+    console.log("In base");
+    console.log("Name: " + nameOfElement);
+    console.log("Element:" + element);
+
+    closeBtn.addEventListener("click", showMethod.bind(this))
+    return this.basis;
+}
+
+function showMethod() {
+    this.show();
+}
+
+function createCloseBtnHTML() {
+    const closeBtn = document.createElement("div");
+    closeBtn.classList.add("close-btn");
+
+    const closeIcon = document.createElement("span");
+    closeIcon.classList.add("fa");
+    closeIcon.classList.add("fa-close");
+    closeBtn.appendChild(closeIcon);
+    return closeBtn;
+}
+
+function showAutomatically() {
+    this.basis.classList.remove("show");
+    this.basis.classList.add("hide");
+}
+
+Base.prototype = {
+    constructor: Base,
+    show: function () {
+        if (this.nameOfElement === "Toast") {
```

Comment:
The parent component should know nothing about his children. Otherwise, you'll have to rework your Base component each time you create new Child component